callbacks/create.go

```diff
@@ -62,46 +62,57 @@ func Create(config *Config) func(db *gorm.DB) {
 				return
 			}
 
-			db.RowsAffected, _ = result.RowsAffected()
+			db.RowsAffected, err = result.RowsAffected()
+			if err != nil {
+				db.AddError(err)
+				return
+			}
 
 			if db.RowsAffected != 0 && db.Statement.Schema != nil &&
 				db.Statement.Schema.PrioritizedPrimaryField != nil && db.Statement.Schema.PrioritizedPrimaryField.HasDefaultValue {
-				if insertID, err := result.LastInsertId(); err == nil && insertID > 0 {
-					switch db.Statement.ReflectValue.Kind() {
-					case reflect.Slice, reflect.Array:
-						if config.LastInsertIDReversed {
-							for i := db.Statement.ReflectValue.Len() - 1; i >= 0; i-- {
-								rv := db.Statement.ReflectValue.Index(i)
-								if reflect.Indirect(rv).Kind() != reflect.Struct {
-									break
-								}
+				insertID, err := result.LastInsertId()
```

Comment:
some driver doesn't support `LastInsertId`, maybe should not return an error here.

Reply:
@jinzhu I have adjusted it, please CR/MR again